sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -318,25 +318,26 @@ export class StudentController extends BaseController {
     }));
   }
   /**
+   * TODO: This api will be called by UI and update states, in future restriction UI ticket
    * GET API which returns student restriction details.
-   * @param userToken
-   * @returns Student Restriction
+   * @param studentToken student token.
+   * @returns Student restriction code and notification type, if any.
    */
   @AllowAuthorizedParty(AuthorizedParties.student)
   @Get("restriction")
   async getStudentRestrictions(
-    @UserToken() userToken: IUserToken,
-  ): Promise<StudentRestrictionDTO> {
-    const studentRestrictionStatus =
-      await this.studentRestrictionService.getStudentRestrictionsByUserId(
-        userToken.userId,
+    @UserToken() studentToken: StudentUserToken,
+  ): Promise<StudentRestrictionAPIOutDTO[]> {
+    const studentRestrictions =
+      await this.studentRestrictionService.getStudentRestrictionsById(
+        studentToken.studentId,
       );
-    return {
-      hasRestriction: studentRestrictionStatus.hasRestriction,
-      hasFederalRestriction: studentRestrictionStatus.hasFederalRestriction,
-      hasProvincialRestriction:
-        studentRestrictionStatus.hasProvincialRestriction,
-      restrictionMessage: studentRestrictionStatus.restrictionMessage,
-    } as StudentRestrictionDTO;
+    if (!studentRestrictions) {
+      return [];
+    }
+    return studentRestrictions.map((studentRestriction) => ({
```

Comment:
I believe that we do just `return studentRestrictions?.map((studentRestriction)`. But it is just a suggestion, it is already good.